src/Data/HeytingAlgebra/Laws.purs

```diff
@@ -0,0 +1,62 @@
+module Data.HeytingAlgebra.Laws where
+
+import Prelude
+
+import Data.HeytingAlgebra (ff, implies, tt)
+import Laws.Algebraic as Laws
+
+-- | `a || (b || c) = (a || b) || c`
+associativeDisj :: forall a. Eq a => HeytingAlgebra a => a -> a -> a -> Boolean
+associativeDisj = Laws.associative (||)
+
+-- | `a && (b && c) = (a && b) && c`
+associativeConj :: forall a. Eq a => HeytingAlgebra a => a -> a -> a -> Boolean
+associativeConj = Laws.associative (&&)
+
+-- | `a || b = b || a`
+commutativeDisj :: forall a. Eq a => HeytingAlgebra a => a -> a -> Boolean
+commutativeDisj = Laws.commutative (||)
+
+-- | `a && b = b && a`
+commutativeConj :: forall a. Eq a => HeytingAlgebra a => a -> a -> Boolean
+commutativeConj = Laws.commutative (&&)
+
+-- | `a || (a && b) = a && (a || b) = a`
+absorption :: forall a. Eq a => HeytingAlgebra a => a -> a -> Boolean
+absorption = Laws.absorption (||) (&&)
+
+-- | `a || a = a`
+idempotentDisj :: forall a. Eq a => HeytingAlgebra a => a -> a -> Boolean
+idempotentDisj = Laws.idempotent (||)
+
+-- | `a && a = a`
+idempotentConj :: forall a. Eq a => HeytingAlgebra a => a -> a -> Boolean
+idempotentConj = Laws.idempotent (&&)
+
+-- | `a || ff = a`
+identityDisj :: forall a. Eq a => HeytingAlgebra a => a -> Boolean
+identityDisj = Laws.identity (||) ff
+
+-- | `a && tt = a`
+identityConj :: forall a. Eq a => HeytingAlgebra a => a -> Boolean
+identityConj = Laws.identity (&&) tt
+
+-- | ``a `implies` a = tt``
+reflexiveImplies :: forall a. Eq a => HeytingAlgebra a => a -> Boolean
+reflexiveImplies a = (a `implies` a) == tt
+
+-- | ``(a && b) `implies` (b && c) = a `implies` c``
```

Comment:
I don’t think we can describe this as transitivity. Usually, saying that a relation ~ is transitive is saying that if A ~ B and B ~ C, then A ~ C. That doesn’t quite match the law here, because && is taking the place of ~ on the left hand side, but `implies` is taking the place of ~ on the right hand side. Also, transitivity doesn’t usually work backwards: knowing that A ~ C doesn’t allow me to deduce that A ~ B and that B ~ C.

Reply:
You're right, I'm not sure what I was doing here. The way I saw this expressed means this should be:

```
((a `implies` b) `implies` (b `implies` c)) `implies` (a `implies` c) = tt
```

Better?

Reply:
I'm not quite sure about this either: don't we want some of those `implies` to be using the `HeytingAlgebra Boolean` instance, rather than having all of them use the `HeytingAlgebra a` instance?